File: String_Reverse/Reverse.c
/*
    Author : Pakshal Shashikant Jain 
    Date : 11/08/2023
    Program : Write a program which accept string from user and display it inn
              reverse order.

              Input : "MarvellouS"
              Output : "SuollevraM"
*/

////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
#include<stdio.h>

////////////////////////////////////////////////////////////////////////////////////////

//Function To Reverse String 
void StrRev(char *str)
{
    //Local Variable 
    int iStart = 0;
    int iEnd = 0;
    char temp = '\0';

    if(str == NULL)
    {
        return;
    }

    //Logic To Reverse The String
    while(str[iEnd] != '\0')
    {
        iEnd++;
    }

    iEnd--;

    while(iStart < iEnd)
    {
        temp = str[iStart];
        str[iStart] = str[iEnd];
        str[iEnd] = temp;

        iStart++;
        iEnd--;
    }
}

/////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
int main()
{
    char arr[20];

    printf("Jay Ganesh.....\n");

    //Taking Input from user
    printf("Enter String You Want \n");
    scanf("%[^'\n']s",arr);

    //Call To StrRev Function
    StrRev(arr);

    //Printing Reverse String 
    printf("Reverse String is : %s",arr);

    return 0;
}

File: String_Reverse/Reverse.go
/*
    Author : Pakshal Shashikant Jain 
    Date : 11/08/2023
    Program : Write a program which accept string from user and display it inn
              reverse order.

              Input : "MarvellouS"
              Output : "SuollevraM"
*/

//Required Libraries
package main

import (
	"fmt"
	"bufio"
	"os"
)

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function To Reverse String 
func StrRev(str []rune) {
	var i int = 0;

	//Logic To Reverse string 
	fmt.Println("Reversed String is : ");
	for i = len(str) - 1;i >= 0 ;i-- {
		fmt.Printf("%c",str[i]);
	} 
}

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
func main() {
	var arr []rune;
	var str string = " ";
	var err error = nil;

	//Creating object of NewReader Class
	var bobj = bufio.NewReader(os.Stdin);

	fmt.Println("Jay Ganesh.....");

	//Taking Input from user
	fmt.Printf("Enter One String\n");
	str,err = bobj.ReadString('\n');

	if(err != nil) {
		fmt.Println("An Error Occured");
	}

	//Creating character of Array of String 
	arr = []rune(str);

	//Call To StrRev Function
	StrRev(arr);
}